### ISSUE TITLE
Rule S930: improve highlighting for no-extra-arguments 

### ISSUE THREAD

**Issue by andrey-tyukin-sonarsource** [SonarSource/eslint-plugin-sonarjs#178]
Highlighting of function invocations with too many arguments can be counter-intuitive if the function call stretches over multiple lines.

Minimal Example:
```
    function functionName(a, b){}
//           ^^^^^^^^^^^^^^^^^^             Secondary location is the function declaration.

    functionName(
      arg1,
      arg2,
      arg3,
      arg4
    );
//  ^ Primary issue location at the end of the call expression 
//     (the closing parenthesis is the last token of the call expression).
```

A real-world example is described and discussed [here](https://discuss.sonarsource.com/t/javascript-missing-highlighting-of-the-primary-location/5537).

The proposed solution is to change the primary location and introduce more secondary locations, as follows:

  * Primary location should be moved to the callee, so that the main error message appears before the opening parenthesis.
  * Every superfluous argument should be reported with a secondary location.
  * If the callee is a single identifier, then this identifier should be used in the error message: *"The function {identifier} expects {N} arguments, but {M} were provided."*; That is, if the call looks like `f(x)`, then `f` should be the primary location.
  * If the callee is a member expression, then the method name should be used in the same message template. That is, if the call looks like `expr.methodName(a1, ..., aN)`, then `expr.methodName` should be the primary location, and `methodName` should be used in the message.
  * In all other cases (where no good name can be found), the original message *"This function expects {N} arguments, but {M} were provided."* should be preserved.

For the example above, it would mean:

```
    function functionName(a, b){}
//           ^^^^^^^^^^^^^^^^^^             Secondary location at the function declaration.

    functionName(
//  ^^^^^^^^^^^^  primary location at the callee
      arg1,
      arg2,
      arg3,
 //   ^^^^ secondary location at superfluous argument
      arg4
 //   ^^^^ secondary location at superfluous argument
    );
```
